src/interpreter_funcs/voxel_interpolated_union.rs

```diff
@@ -0,0 +1,260 @@
+                "Too many voxels. Limit set to {}. Try setting voxel size to [{:.3}, {:.3}, {:.3}] or more.",
+                max_count, x, y, z
+            ),
+        }
+    }
+}
+
+impl error::Error for FuncInterpolatedUnionError {}
+
+pub struct FuncInterpolatedUnion;
+
+impl Func for FuncInterpolatedUnion {
+    fn info(&self) -> &FuncInfo {
+        &FuncInfo {
+            name: "Voxel Morph",
+            description: "VOXEL INTERPOLATION MORPH FROM TWO MESH GEOMETRIES\n\
+            \n\
+            Converts the input mesh geometries into voxel clouds, resizes both voxel \
+            clouds to bel large enough to contain volumes from both of them \
+            then computes distance fields for both voxel clouds, interpolates values \
+            of the two distance fields by the given factor and eventually \
+            materializes the resulting voxel cloud into a welded mesh.\n\
+            \n\
+            Voxels are three-dimensional pixels. They exist in a regular three-dimensional \
+            grid of arbitrary dimensions (voxel size). The voxel can be turned on \
+            (be a volume) or off (be a void). The voxels can be materialized as \
+            rectangular blocks. Voxelized meshes can be effectively smoothened by \
+            Laplacian relaxation.
+            \n\
+            The input meshes will be marked used and thus invisible in the viewport. \
+            They can still be used in subsequent operations.\n\
+            \n\
+            The resulting mesh geometry will be named 'Interpolated Mesh'.",
+            return_value_name: "Interpolated Mesh",
+        }
+    }
+
+    fn flags(&self) -> FuncFlags {
+        FuncFlags::PURE
+    }
+
+    fn param_info(&self) -> &[ParamInfo] {
+        &[
+            ParamInfo {
+                name: "Mesh 1",
+                description: "First input mesh.",
+                refinement: ParamRefinement::Mesh,
+                optional: false,
+            },
+            ParamInfo {
+                name: "Mesh 2",
+                description: "Second input mesh.",
+                refinement: ParamRefinement::Mesh,
+                optional: false,
+            },
+            ParamInfo {
+                name: "Voxel Size",
+                description: "Size of a single cell in the regular three-dimensional voxel grid.\n\
+                \n\
+                High values produce coarser results, low values may increase precision but produce \
+                heavier geometry that significantly affect performance. Too high values produce \
+                single large voxel, too low values may generate holes in the resulting geometry.",
+                refinement: ParamRefinement::Float3(Float3ParamRefinement {
+                    min_value: Some(0.005),
+                    max_value: None,
+                    default_value_x: Some(0.25),
+                    default_value_y: Some(0.25),
+                    default_value_z: Some(0.25),
```

Comment:
we need to decide. if the uses plays with the default primitives, which are always around 1 unit big, it may make sense to set the default voxel size to somewhere around 0.1 (I know it's not 0.25).
on the other hand, when the primitives are by default set to the unit size (1), then also the voxels could be set to 1. in real life though, many users will get frustrated when playing with the software and learning, that they get visually unpleasing results with the default values.
on the third hand, most of the imported models are made in mm (products) or cm (german  world) or m (architecture and urban planning) and the objects are usually several thousands of units big. in such a case, even voxel size 1 is too small to perform well. 

any suggestion?

Reply:
Tough question... Maybe let's optimize for our current imported models (meaning real units that come in the hundreds and thousands) and keep this at 1 for now? We will have to increase the safe voxel limit for this so that this doesn't always error.

Reply:
regarding the voxel limit - can I put it somewhere central/global?

Reply:
also, what about adjusting the primitives to roughly match the scale of our sample OBJs?

like setting them to 10, 100 or 1000 even by default

Reply:
> like setting them to 10, 100 or 1000 even by default

Yeah, let's do that (in another PR)